eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -661,13 +661,22 @@ object Helpers {
     def claimRemoteCommitMainOutput(keyManager: KeyManager, commitments: Commitments, remotePerCommitmentPoint: PublicKey, tx: Transaction, feeEstimator: FeeEstimator, feeTargets: FeeTargets)(implicit log: LoggingAdapter): RemoteCommitPublished = {
       val channelKeyPath = keyManager.channelKeyPath(commitments.localParams, commitments.channelVersion)
       val localPubkey = Generators.derivePubKey(keyManager.paymentPoint(channelKeyPath).publicKey, remotePerCommitmentPoint)
+      val localPaymentPoint = keyManager.paymentPoint(channelKeyPath).publicKey
       val feeratePerKwMain = feeEstimator.getFeeratePerKw(feeTargets.claimMainBlockTarget)
 
-      val mainTx = generateTx("claim-p2wpkh-output") {
-        Transactions.makeClaimP2WPKHOutputTx(tx, commitments.localParams.dustLimit, localPubkey, commitments.localParams.defaultFinalScriptPubKey, feeratePerKwMain).right.map(claimMain => {
-          val sig = keyManager.sign(claimMain, keyManager.paymentPoint(channelKeyPath), remotePerCommitmentPoint, TxOwner.Local, commitments.commitmentFormat)
-          Transactions.addSigs(claimMain, localPubkey, sig)
-        })
+      val mainTx = commitments.commitmentFormat match {
+        case DefaultCommitmentFormat => generateTx("claim-p2wpkh-output") {
+          Transactions.makeClaimP2WPKHOutputTx(tx, commitments.localParams.dustLimit, localPubkey, commitments.localParams.defaultFinalScriptPubKey, feeratePerKwMain).right.map(claimMain => {
+            val sig = keyManager.sign(claimMain, keyManager.paymentPoint(channelKeyPath), remotePerCommitmentPoint, TxOwner.Local, commitments.commitmentFormat)
+            Transactions.addSigs(claimMain, localPubkey, sig)
+          })
+        }
+        case AnchorOutputsCommitmentFormat => generateTx("claim-remote-delayed-output") {
+          Transactions.makeClaimRemoteDelayedOutputTx(tx, commitments.localParams.dustLimit, localPaymentPoint, commitments.localParams.defaultFinalScriptPubKey, feeratePerKwMain).right.map(claimMain => {
```

Comment:
Should we rename `Transactions.makeClaimDelayedOutputTx` to `Transactions.makeClaimLocalDelayedOutputTx` for consistency? Those scripts are very difficult to identify and navigate, not to mention the fact that commitments are symmetric.

Reply:
I agree, I'm wondering if it's worth renaming that now or in a refactor PR, but those in separate objects to make it even clearer (we can also rename now, and refactor further later).

Reply:
Done in https://github.com/ACINQ/eclair/pull/1501/commits/d140f8bee908e287f5a6cab5888dbc3aae71846e